src/sparql/PatternBuilder.ts

```diff
@@ -15,6 +15,57 @@ export default class PatternBuilder {
 	public buildValuePatternFromCondition( condition: Condition, conditionIndex: number ): Pattern[] {
 		const negate: boolean = condition.negate || false;
 
+		const bgp: Pattern = {
+			type: 'bgp',
+			triples: [ this.tripleBuilder.buildTripleFromQueryCondition( condition, conditionIndex ) ],
+		};
+
+		const patterns: Pattern[] = [];
+
+		if ( negate === true ) {
+			patterns.push( {
+				type: 'minus',
+				patterns: [ bgp ],
+			} );
+		} else {
+			if ( condition.referenceRelation === ReferenceRelation.With ||
+				condition.referenceRelation === ReferenceRelation.Without ) {
```

Comment:
can't we just check `regardless` here ?
`if (condition.referenceRelation !== ReferenceRelation.regardless)`

Reply:
I just moved this around, but you're right, comparing only with regardless is easier to understand.

Done.